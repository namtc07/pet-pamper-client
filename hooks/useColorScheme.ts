// eslint-disable-next-line import/prefer-default-export
export { useColorScheme } from 'react-native';
