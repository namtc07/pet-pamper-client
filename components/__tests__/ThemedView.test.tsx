import React from 'react';
import { render } from '@testing-library/react-native';
import { useThemeColor } from '@/hooks/useThemeColor';
import ThemedView from '../ThemedView';

// Mock useThemeColor hook
jest.mock('@/hooks/useThemeColor', () => ({
  useThemeColor: jest.fn(),
}));

describe('ThemedView Snapshot Tests', () => {
  it('should match snapshot with light color', () => {
    // Mocking useThemeColor hook to return a specific color
    (useThemeColor as jest.Mock).mockReturnValue('white');

    const { toJSON } = render(
      <ThemedView lightColor="white" darkColor="black" />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with dark color', () => {
    // Mocking useThemeColor hook to return a specific color
    (useThemeColor as jest.Mock).mockReturnValue('black');

    const { toJSON } = render(
      <ThemedView lightColor="white" darkColor="black" />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with additional styles and props', () => {
    (useThemeColor as jest.Mock).mockReturnValue('blue');

    const { toJSON } = render(
      <ThemedView
        lightColor="white"
        darkColor="black"
        style={{ borderWidth: 1 }}
        accessibilityLabel="Themed View"
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
