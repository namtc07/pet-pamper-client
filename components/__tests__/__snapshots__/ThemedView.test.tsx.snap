// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedView Snapshot Tests should match snapshot with additional styles and props 1`] = `
<View
  accessibilityLabel="Themed View"
  style={
    [
      {
        "backgroundColor": "blue",
      },
      {
        "borderWidth": 1,
      },
    ]
  }
/>
`;

exports[`ThemedView Snapshot Tests should match snapshot with dark color 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "black",
      },
      undefined,
    ]
  }
/>
`;

exports[`ThemedView Snapshot Tests should match snapshot with light color 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
      },
      undefined,
    ]
  }
/>
`;
